demo/grocerystore/grocerystore.sysl

```diff
@@ -0,0 +1,125 @@
+GroceryStore [package="GroceryStore", ~rest]:
+    @owner.name = "Eddard Stark"
+    @team.slack = "https://slack.com/intl/en-au/"
+    @repo.url = "https://github.com/anz-bank/sysl"
+    @docs.url = "https://en.wikipedia.org/wiki/Grocery_store"
+    @description =:
+        | The Grocery Store API lets users shop for groceries.
+    @version = "1.0.0"
+    !type InventoryResponse[json_map_key="item_id"]:
+        item_id <: string:
+            @json_tag = "item_id"
+        quantity <: int:
+            @json_tag = "quantity"
+
+    !type CartRequest:
+        item_id <: string:
+            @json_tag = "item_id"
+            @rpcId = "5"
+        cart_id <: string:
+            @json_tag = "cart_id"
+            @rpcId = "6"
+
+    !type CartResponse:
+        transaction_id <: string[~header]:
+            @json_tag = "transaction_id"
+            @rpcId = "1"
+        item_id <: string[~body]:
+            @json_tag = "item_id"
+            @rpcId = "2"
+        cart_id <: string:
+            @json_tag = "cart_id"
+            @rpcId = "4"
+
+    /inventory:
+        GET(fooid <: string [~header, ~required, name="FooID"]):
+            | Returns a list of items in the inventory
+            Inventory <- GET /inventory
+            return sequence of InventoryResponse
+
+
+    /cart:
+        GET:
+            | Adds the specified item to the cart
+            Cart <- GET /cart
+            return CartResponse
+        POST:
+            | Adds the specified item to the cart
+            Cart <- POST /cart
+            return CartResponse
+        DELETE:
+            | Removes the specified item from the cart
+            Cart <- DELETE /cart
+            return CartResponse
+
+    /checkout:
+        POST?payment_info=string: 
+            Payment <- POST /validate
+            Payment <- POST /pay
+            | Checks out the specified cart
+            return ok <: string
+
+Inventory[package="InventoryService", ~rest]:
+    @version = "1.0.0"
+    @owner.name = "Jon Snow"
+    @team.slack = "https://slack.com/intl/en-au/"
+    @repo.url = "https://github.com/anz-bank/sysl"
+    @docs.url = "https://en.wikipedia.org/wiki/Grocery_store"
+    @description =:
+        | The Inventory API gives information on items available for purchase.
+
+    /inventory:
+        GET:
+            | Returns a list of available items for purchase 
+            return ok <: string
+
+Cart[package="InventoryService", ~rest]:
+    @owner.name = "Sam Tarwell"
```

Comment:
Samwell Tarly? Was this intentional? xD

Reply:
hahaha yeah, I forgot most of the character names so some might be wrong. Thanks for that pickup